models/purchase_order.py

```diff
@@ -14,6 +14,34 @@ def button_approve(self):
         res = super(PurchaseOrder, self).button_approve()
         return res
 
+    invoice_payment_status = fields.Selection([
+          ('no', 'Nothing to Pay'),
+          ('not_paid', 'Not Paid'),
+          ('in_payment', 'In Payment'),
+          ('paid', 'Paid'),
+          ('partially_paid', 'Partially Paid'),
+      ], string='Invoice Payment Status', compute='_get_invoice_payement_status', store=True, readonly=True)
+
+    @api.depends('state', 'order_line.invoice_lines.move_id.payment_state')
+    def _get_invoice_payement_status(self):
```

Comment:
@MaganaAsiati 
Small typo here: '_get_invoice_payment_status' instead of '_get_invoice_payement_status'